ms_active_directory/core/ad_objects.py

```diff
@@ -170,6 +170,14 @@ def __repr__(self):
     def __str__(self):
         return self.__repr__()
 
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return other.distinguished_name == self.distinguished_name and other.domain == self.domain
```

Comment:
domain objects might not be equivalent if different sites are used, even if the domain is really the same. but things like domain SID can't be fetched without a session. so I'll do some thinking on the best way to handle this. for now, this works for single-domain-object setups, but not as well if an application is multi-tenant and spans a number of networks